Sources/TuistGraph/Graph/ConditionalGraphTarget.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import TSCBasic
+
+public struct GraphTargetReference: Equatable, Comparable, Hashable, CustomDebugStringConvertible, CustomStringConvertible,
```

Comment:
As a naming idea, what do you think of `GraphTargetWithConditions`, still near enough to `GraphTarget` to show its similarity but also verbose enough to communicate what it stands for ?